icecream/icecream.py

```diff
@@ -154,15 +156,39 @@ def format_pair(prefix, arg, value):
     return '\n'.join(lines)
 
 
-def argumentToString(obj):
-    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj)
+def argumentToString(obj, width=DEFAULT_LINE_WRAP_WIDTH):
+    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj, width=width)
     s = s.replace('\\n', '\n')  # Preserve string newlines in output.
     return s
 
 
+def columns():
+    """ Returns the number of columns that this terminal can handle. """
+    width = DEFAULT_LINE_WRAP_WIDTH
+    try:
+        # TODO come up with a more elegant solution than subtracting
+        #  a seemingly random number of characters.
+        width = os.get_terminal_size().columns - COLUMN_OVERFLOW
+    except OSError:  # Not in TTY
+        pass
+    except AttributeError:  # Python 2.x
```

Comment:
This looks a bit too broad and prone to hiding real errors. I'd prefer either checking the Python version or checking for the presence of the relevant attribute.

Reply:
Done.